charts/prometheus-node-exporter/templates/daemonset.yaml

```diff
@@ -24,8 +24,8 @@ spec:
 {{- if and .Values.rbac.create .Values.serviceAccount.create }}
       serviceAccountName: {{ template "prometheus-node-exporter.serviceAccountName" . }}
 {{- end }}
-{{- if .Values.securityContext }}
-      securityContext: {{ toYaml .Values.PodsecurityContext | nindent 8 }}
+{{- with .Values.podSecurityContext }}
```

Comment:
What is the value of switching from `if` to `with` ? Is it just cleaner? Is the functionality the same and the produced output is the same?

Reply:
with allows to completely turn off the config by passing podSecurityContext: [] this means an empy values

Reply:
with the existing if condition it will work like below 
alertmanager: 
  PodSecurityContext: {}
  
 i can revert the code and re-apply the similar config  to cross verify the same